sdk/resource/benchmark_test.go

```diff
@@ -76,3 +79,140 @@ func BenchmarkMergeResource_8(b *testing.B) {
 func BenchmarkMergeResource_16(b *testing.B) {
 	benchmarkMergeResource(b, 16)
 }
+
+type instantDetector struct{}
+
+// instant detector don't do anything
+// its benchmark the overhead of the resource.New implementation.
+func (f instantDetector) Detect(_ context.Context) (*resource.Resource, error) {
+	return resource.NewSchemaless(), nil
+}
+
+type fastDetector struct{}
+
+func (f fastDetector) Detect(_ context.Context) (*resource.Resource, error) {
+	time.Sleep(time.Millisecond)
+	return resource.NewSchemaless(), nil
+}
+
+type mediumDetector struct{}
+
+func (f mediumDetector) Detect(_ context.Context) (*resource.Resource, error) {
+	time.Sleep(time.Millisecond * 30)
+	return resource.NewSchemaless(semconv.ServerAddress("localhost")), nil
+}
+
+type slowDetector struct{}
```

Comment:
This could be changed into a single detector:

```golang
type testDetector struct{
    waitTime time.Duration
}

func (td testDetector) Detect(_ context.Context) (*resource.Resource, error) {
	time.Sleep(td.waitTime * time.Millisecond)
	return resource.NewSchemaless(semconv.ServerAddress("localhost")), nil
}
```


```golang
func BenchmarkNewResourceOverHead_1(b *testing.B) {
	benchmarkOverhead(context.Background(), b, testDetector{0}, 1)
}

func BenchmarkFastDetector_1(b *testing.B) {
	benchmarkOverhead(context.Background(), b, testDetector{1}, 1)
}
```

Also, using subtests rather than creating many benchmarks would make things more concise.